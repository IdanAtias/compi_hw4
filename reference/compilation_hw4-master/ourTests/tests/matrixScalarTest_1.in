int a = 3;
int b = 4, c = -2;

matrix m1(3,2) = [-2,3;4,-7;0,5], m2(4,4) = [1,2,3,4;-2,-5,1,3;7,6,5,4;2,1,3,6];

a * m1;
m1 * a;
(a + 2) * m2;
m1 * (a + b);
b / c * m1 * 5;
(6 / 3) * m2;
matrix m3(3,2) = (a + 3) * m1 * 7;

m2 ^ (3);

[-2,3,-1;4,-7,4;0,5,2] ^ (5);
transpose([1,3;4,-5]) ^ (2);
[1;-2]::[3;5];
matrix mmm4(2,2) = transpose([1;-2]::[3;5]) ^ (4);
[-2,3,-1;4,-7,4;0,5,2] * (6 / 2);