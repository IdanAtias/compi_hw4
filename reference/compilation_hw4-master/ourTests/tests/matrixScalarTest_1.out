a=
3
b=
4
c=
-2
m1=
-2,3
4,-7
0,5
m2=
1,2,3,4
-2,-5,1,3
7,6,5,4
2,1,3,6
-6,9
12,-21
0,15
-6,9
12,-21
0,15
5,10,15,20
-10,-25,5,15
35,30,25,20
10,5,15,30
-14,21
28,-49
0,35
20,-30
-40,70
0,-50
2,4,6,8
-4,-10,2,6
14,12,10,8
4,2,6,12
m3=
-84,126
168,-294
0,210
314,220,390,550
-20,-91,105,180
630,460,722,1002
385,250,499,715
-16912,35012,-12768
41196,-84867,31320
-16460,35035,-11928
13,-16
-12,37
1,3
-2,5
mmm4=
-191,-168
252,145
-6,9,-3
12,-21,12
0,15,6
